### PR TITLE
fix position total value display

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-####

## What changed (plus any additional context for devs)

some positions are omitted from the total wallet balance, but should still display the position value in the position card

## Screen recordings / screenshots
before/after

Screenshot 2024-10-09 at 20 14 44

Screenshot 2024-10-09 at 20 14 48

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1920-display-moxie-positions-usd-value-in-wallet-correctly/1b40042fcc6990b474d0489583c1a72d43e873a4.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1920-display-moxie-positions-usd-value-in-wallet-correctly/1b40042fcc6990b474d0489583c1a72d43e873a4.ipa) for 1b40042fcc6990b474d0489583c1a72d43e873a4

File: src/resources/defi/utils.ts [hunk 1]
```diff
@@ -156,22 +168,30 @@ export const parsePositions = (data: AddysPositionsResponse, currency: NativeCur
...
-      total: totalDisplay,
+      total: {
+        amount: totalAmount,
+        display: convertAmountToNativeDisplay(totalAmount, currency),
+      },
+      /**
+       * omittedTotal is the total WITHOUT the positions that are omitted
+       * used for the total wallet balance, the position card still shows the normal total
+       */
+      omittedTotal: {
```
Comment (greg-schrammel) [inline src/resources/defi/utils.ts:191]: found this name (`ommitedTotal`) kinda confusing but couldn't think of anything better, lmk if you think of a more descriptive name

File: src/resources/defi/utils.ts [hunk 2]
```diff
@@ -52,10 +56,13 @@ export const parsePosition = (position: Position, currency: NativeCurrencyKey):
           underlying.asset.price?.value!,
           currency
         );
-        if (!borrow.omit_from_total) {
-          totalBorrows = add(totalBorrows, nativeDisplay.amount);
+
+        if (borrow.omit_from_total) {
+          totalOmittedFromTotals = subtract(totalOmittedFromTotals, nativeDisplay.amount);
```
Comment (greg-schrammel) [inline src/resources/defi/utils.ts:61]: this make sense? we add the others but subtract the borrows

Comment (greg-schrammel) [reply src/resources/defi/utils.ts:61]: I mean it works right, not sure if I missed some edge case

**Comment by derHowie** [comment]
@greg-schrammel, code looks good but I'm unsure what to test. Is the moxie case the only thing I should test or are there other considerations?

**Comment by greg-schrammel** [comment]
> @greg-schrammel, code looks good but I'm unsure what to test. Is the moxie case the only thing I should test or are there other considerations?

yeah the only positions I know that can have the `omit_from_total ` flag rn are moxie

**Review by walmat — APPROVED** [review]

File: src/resources/defi/utils.ts (hunk 1 above)
Comment (benisgold) [reply src/resources/defi/utils.ts:191]: i think it would be a lot more clear if we first came up for a name for the types of positions that are being excluded. i'm still unclear on this. what dequalifies a position from having it's value being included in the total wallet balance?

**Review by benisgold — APPROVED** [review]
